core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamProcessingStrategy.java

```diff
@@ -210,4 +220,136 @@ public final void dispose() {
       innerSink.dispose();
     }
   }
+
+  private final class RetrySchedulerWrapper implements Scheduler {
+
+    private Scheduler delegate;
+
+    public RetrySchedulerWrapper(Scheduler scheduler) {
+      this.delegate = scheduler;
+    }
+
+    @Override
+    public ScheduledFuture<?> scheduleWithCronExpression(Runnable command, String cronExpression) {
+      return delegate.scheduleWithCronExpression(command, cronExpression);
+    }
+
+    @Override
+    public ScheduledFuture<?> scheduleWithCronExpression(Runnable command, String cronExpression, TimeZone timeZone) {
+      return delegate.scheduleWithCronExpression(command, cronExpression, timeZone);
+    }
+
+    @Override
+    public void stop() {
+      delegate.stop();
+    }
+
+    @Override
+    public String getName() {
+      return delegate.getName();
+    }
+
+    @Override
+    public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+      return delegate.schedule(command, delay, unit);
+    }
+
+    @Override
+    public <V> ScheduledFuture<V> schedule(Callable<V> callable, long delay, TimeUnit unit) {
+      return delegate.schedule(callable, delay, unit);
+    }
+
+    @Override
+    public ScheduledFuture<?> scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit) {
+      return delegate.scheduleAtFixedRate(command, initialDelay, period, unit);
+    }
+
+    @Override
+    public ScheduledFuture<?> scheduleWithFixedDelay(Runnable command, long initialDelay, long delay, TimeUnit unit) {
+      return delegate.scheduleWithFixedDelay(command, initialDelay, delay, unit);
+    }
+
+    @Override
+    public void shutdown() {
+      delegate.shutdown();
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+      return delegate.shutdownNow();
+    }
+
+    @Override
+    public boolean isShutdown() {
+      return delegate.isShutdown();
+    }
+
+    @Override
+    public boolean isTerminated() {
+      return delegate.isTerminated();
+    }
+
+    @Override
+    public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
+      return delegate.awaitTermination(timeout, unit);
+    }
+
+    private <T> Future<T> submit(Supplier<Future<T>> realSubmit) {
+      while (!this.isShutdown() && !this.isTerminated()) {
+        try {
+          return realSubmit.get();
+        } catch (RejectedExecutionException ree) {
+          lastRetryTimestamp.set(nanoTime());
+          try {
+            Thread.sleep(SCHEDULER_BUSY_RETRY_INTERVAL_MS);
+          } catch (InterruptedException e) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2142))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7624&issues=AWh8XXbHotIXnmsUOEzB&open=AWh8XXbHotIXnmsUOEzB)

[ISSUE_KEY]: AWh8XXbHotIXnmsUOEzB